### PR TITLE
[CO-2720] Added catalog liquid templating document

### PR DESCRIPTION
**Pull request by shihanxiong**
https://jira.braze.com/browse/CO-2720

Screenshot 2024-06-24 at 1 33 15 PM

### CODE REVIEW

**Comment by shihanxiong** [comment]
@lydia-xie FYI we want this change to be gated for now. cc @johndinh

File: _docs/_user_guide/personalization_and_dynamic_content/catalogs/catalog.md [hunk 1]
```diff
@@ -320,6 +320,34 @@ Which will display as the following:
 
 With templating, you can render a different catalog item for each user based on their individual custom attributes, event properties, or any other templatable field.
 
+#### Templating catalog items that contain liquid
```
Comment (davehensley) [inline _docs/_user_guide/personalization_and_dynamic_content/catalogs/catalog.md:323]: Nit: I think we usually capitalize "Liquid".

**Review by davehensley — APPROVED** [review]

**Review by HVisagie — APPROVED** [review]

File: _docs/_user_guide/personalization_and_dynamic_content/catalogs/catalog.md [hunk 2]
```diff
@@ -328,10 +328,6 @@ You can upload a CSV of new catalog items to add or catalog items to update. To
 
 You can also manually piece together catalogs Liquid logic. However, note that if you type in an ID that doesn't exist, Braze will still return an items array without objects. We recommend that you include error handling, such as checking the size of the array and using an `if` statement to account for an empty array case.
 
-{% alert note %}
-Liquid currently can't be used inside catalogs. If Liquid personalization is listed inside a cell in your catalog, the dynamic value won't render and only the actual Liquid will display.
-{% endalert %}
```
Comment (shihanxiong) [inline _docs/_user_guide/personalization_and_dynamic_content/catalogs/catalog.md:333]: @lydia-xie FYI we also will be deleting this section as part of this PR

**Review by lydia-xie — APPROVED** [review]
lgtm!

**Pull request merged**